src/gui/plugins/video_recorder/VideoRecorder.qml

```diff
@@ -37,16 +37,67 @@ ToolBar {
         title: "Save the recorded video"
         folder: shortcuts.home
         selectExisting: false
+        property var fileFormat: ""
+        property var selectedFormat: ""
+
+        function getFormat(url) {
+          return (url.slice(url.lastIndexOf(".") + 1))
+        }
+        
         onAccepted: {
-          VideoRecorder.OnSave(fileDialog.fileUrl)
-          close()
+          fileFormat = getFormat(fileUrl.toString())
+          if (fileFormat == fileUrl.toString()) {
+            // no format specified
+            VideoRecorder.OnSave(fileFormat + "." + selectedFormat)
+            close()
+          }
+          else if (fileFormat != selectedFormat){
+            // wrong format specified
+            mismatchDialog.open()
+          } else {
+            // correct format specified
+            VideoRecorder.OnSave(fileUrl)
+            close()
+          }
         }
         onRejected: {
           VideoRecorder.OnCancel()
           close()
         }
     }
 
+    Dialog {
+      id: mismatchDialog 
+      title: "Enconding and filename mismatch"
+      modal: true
+      focus: false
+      width: 700
+      height: 200
+      parent: ApplicationWindow.overlay
+      x: (parent.width - width) / 2
+      y: (parent.height - height) / 2
+      standardButtons: Dialog.Save | Dialog.Abort
+
+      GridLayout {
+        id: grid
+        rows: 2
+        flow: GridLayout.TopToBottom
+        anchors.fill: parent
+        Label { text: "File extension doesn't correspond with video encoding."; font.pixelSize: 20 }
```

Comment:
Not sure about our style guide for QML, but I have a feeling the things in the braces `{}` should be on a new line, and then one line per field like the other existing items.

Reply:
Thanks! Yes, I think so.